src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/InlineCompletion/InlineCompletionEndPoint.cs

```diff
@@ -0,0 +1,240 @@
+            {
+                continue;
+            }
+
+            var remappedItem = new InlineCompletionItem
+            {
+                Command = item.Command,
+                Range = rangeInRazorDoc,
+                Text = newSnippetText.ToString(),
+                TextFormat = item.TextFormat,
+            };
+            items.Add(remappedItem);
+        }
+
+        _logger.LogInformation($"Returning items.");
+        return new InlineCompletionList
+        {
+            Items = items.ToArray()
+        };
+    }
+
+    private static bool TryGetSnippetWithAdjustedIndentation(FormattingContext formattingContext, string snippetText, int hostDocumentIndex, [NotNullWhen(true)] out string? newSnippetText)
+    {
+        newSnippetText = null;
+        if (!formattingContext.TryGetFormattingSpan(hostDocumentIndex, out var formattingSpan))
+        {
+            return false;
+        }
+
+        // Take the amount of indentation razor and html are adding, then remove the amount of C# indentation that is 'hidden'.
+        // This should give us the desired base indentation that must be applied to each line.
+        var razorAndHtmlContributionsToIndentation = formattingSpan.RazorIndentationLevel + formattingSpan.HtmlIndentationLevel;
+        var amountToAddToCSharpIndentation = razorAndHtmlContributionsToIndentation - formattingSpan.MinCSharpIndentLevel;
+
+        var snippetSourceText = SourceText.From(snippetText);
+        List<TextChange> indentationChanges = new();
+        // Adjust each line, skipping the first since it must start at the snippet keyword.
+        foreach (var line in snippetSourceText.Lines.Skip(1))
+        {
+            var text = snippetSourceText.GetSubText(line.Span).ToString();
+            if (string.IsNullOrEmpty(text))
+            {
+                // We just have an empty line, nothing to do.
+                continue;
+            }
+
+            // Get the indentation of the line in the C# document based on what options the C# document was generated with.
+            var csharpLineIndentationSize = line.GetIndentationSize(formattingContext.Options.TabSize);
+            var csharpIndentationLevel = csharpLineIndentationSize / formattingContext.Options.TabSize;
+
+            // Get the new indentation level based on the context in the razor document.
+            var newIndentationLevel = csharpIndentationLevel + amountToAddToCSharpIndentation;
+            var newIndentationString = formattingContext.GetIndentationLevelString(newIndentationLevel);
+
+            // Replace the current indentation with the new indentation.
+            var spanToReplace = new TextSpan(line.Start, line.GetFirstNonWhitespaceOffset() ?? line.Span.End);
+            var textChange = new TextChange(spanToReplace, newIndentationString);
+            indentationChanges.Add(textChange);
```

Comment:
Can't we just add an edit to the start of each line, that inserts the restult of `formattingContext.GetIndentationLevelString(newIndentationLevel)` at char 0? Why bother detecting current c# indentation of the snippet? If we're assuming they're formatted correctly, it seems fair to assume they're formatted relative to column 0.

Reply:
Discussed offline, can't do this.